IPython/extensions/autoreload.py

```diff
@@ -500,6 +500,11 @@ def aimport(self, parameter_s='', stream=None):
             else:
                 stream.write("Modules to reload:\n%s\n" % ' '.join(to_reload))
             stream.write("\nModules to skip:\n%s\n" % ' '.join(to_skip))
+        elif isinstance(modname.split(" "), list):
```

Comment:
`str.split()` will always return a list, even if you call it with an empty string. So this check is redundant. It does need some checking, however, because as it stands, you'll never end up in the branch below for `-foo`.